contributors/design-proposals/cluster-info-configmap.md

```diff
@@ -0,0 +1,25 @@
+# Persist cluster-wide data in ConfigMap
+
+## Abstract
+
+We want to allow cluster-wide data to be persisted for general purposes, currently for us (service catalog) is to focus on the ability to identify different clusters that access our service broker with a 'cluster-id'.
+
+## Motivation
+
+Currently there is no way for service catalog api server to identify the different clusters that access our service brokers. The goal of this design is to have a unify storage for cluster related information. The information should be retained and survive between component failures. 
+
+## Detailed design
+
+A configmap `cluster-info` is created in the `default` namespace at the starup of the controller. The initial content of the map will be as the following:
```

Comment:
We should also be careful not to tie ourselves to a ConfigMap for exposing `cluster-info`, and I'm also not sure we should allow components to rely *directly* on the config source object. There can be a place you push config to, and a place you consume it from, and they don't have to be the same. We may want to change the source type later, so keeping these things decoupled is important. One way would be to have Kubelets consume the `cluster-info` source and plumb it through the downward API.